### PR TITLE
RHTAP-3027 Jenkins/Github advanced test with promotion pipeline

### PR DESCRIPTION
**Pull request by jkopriva**

### CODE REVIEW

**Review by rhopp — CHANGES_REQUESTED** [review]

File: tests/gpts/github/test-config/github_advanced_jenkins.ts [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+                    await new Promise(resolve => setTimeout(resolve, 5000));
+                    const jobStatus = await jenkinsClient.waitForBuildToFinish(repositoryName, 1);
+                    expect(jobStatus).not.toBe(undefined);
+                    expect(jobStatus).toBe("SUCCESS");
+                }, 240000);
+
+        /**
+        * Trigger a promotion Pull Request in Gitops repository to promote stage image to prod environment
+        */
+        it('trigger pull request promotion to promote from stage to prod environment', async () => {
```
[the remark this answers is not in the document]
Comment (jkopriva) [reply tests/gpts/github/test-config/github_advanced_jenkins.ts]: Fixed

File: tests/gpts/github/test-config/github_advanced_jenkins.ts [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+         */
+        it(`merge gitops pull request to sync new image in stage environment`, async () => {
+            await gitHubClient.mergePullRequest(githubOrganization, `${repositoryName}-gitops`, gitopsPromotionPRNumber)
+        }, 120000)
+
+
+        /**
+         * Trigger and wait for Jenkins job to finish(it will also run deplyment pipeline)
+         */
+        it(`Trigger job and wait for ${gptTemplate} jenkins job to finish`, async () => {
```
[the remark this answers is not in the document]
Comment (jkopriva) [reply tests/gpts/github/test-config/github_advanced_jenkins.ts]: Fixed

File: tests/gpts/github/test-config/github_advanced_jenkins.ts [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+         * Merge the gitops Pull Request with the new image value. Expect that argocd will sync the new image in stage 
+         */
+        it(`merge gitops pull request to sync new image in stage environment`, async () => {
+            await gitHubClient.mergePullRequest(githubOrganization, `${repositoryName}-gitops`, gitopsPromotionPRNumber)
+        }, 120000)
+
+        /**
+        * Trigger and wait for Jenkins job to finish(it will also run deplyment pipeline)
+        */
+        it(`Trigger job and wait for ${gptTemplate} jenkins job to finish`, async () => {
```
[the remark this answers is not in the document]
Comment (jkopriva) [reply tests/gpts/github/test-config/github_advanced_jenkins.ts]: Fixed

File: tests/gpts/github/test-config/github_advanced_jenkins.ts [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+                gitopsPromotionPRNumber = gitopsPromotionPR
+            } else {
+                throw new Error("Failed to create a pr");
+            }
+        })
+
+        /**
+         * Merge the gitops Pull Request with the new image value. Expect that argocd will sync the new image in stage 
+         */
+        it(`merge gitops pull request to sync new image in stage environment`, async () => {
```
[the remark this answers is not in the document]
Comment (jkopriva) [reply tests/gpts/github/test-config/github_advanced_jenkins.ts]: Fixed

File: tests/gpts/github/test-config/github_advanced_jenkins.ts [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+            await new Promise(resolve => setTimeout(resolve, 5000));
+            const jobStatus = await jenkinsClient.waitForBuildToFinish(repositoryName + "-gitops", 3);
+            expect(jobStatus).not.toBe(undefined);
+            expect(jobStatus).toBe("SUCCESS");
+        }, 240000);
+
+        /**
+         * Obtain the openshift Route for the component and verify that the previous builded image was synced in the cluster and deployed in prod environment
+         */
+        it('container component is successfully synced by gitops in prod environment', async () => {
```
[the remark this answers is not in the document]
Comment (jkopriva) [reply tests/gpts/github/test-config/github_advanced_jenkins.ts:296]: Fixed

**Comment by jkopriva** [comment]
@rhopp Can you please review once again?
image

File: src/apis/git-providers/github.ts [hunk 6]
```diff
@@ -215,6 +216,92 @@ export class GitHubProvider extends Utils {
...
 
+        /**
+     * Enables ACS scan for testing to the main branch of a specified Git repository.
+     * 
+     * @param {string} gitOrg - The name of the GitHub organization.
+     * @param {string} gitRepository - The name of the repository where the file will be committed.
+     * @returns {Promise<string | undefined>} A Promise resolving to the SHA of the commit if successful, otherwise undefined.
+     * @throws Any error that occurs during the execution of the function.
+     */
+        public async updateTUFMirror(gitOrg: string, gitRepository: string, tufURL: string): Promise<string | undefined> {
```
Comment (rhopp) [inline src/apis/git-providers/github.ts:227]: Indentation is off

File: src/apis/git-providers/github.ts [hunk 7]
```diff
@@ -215,6 +216,92 @@ export class GitHubProvider extends Utils {
...
+
+                /**
+     * Enables ACS scan for testing to the main branch of a specified Git repository.
+     * 
+     * @param {string} gitOrg - The name of the GitHub organization.
+     * @param {string} gitRepository - The name of the repository where the file will be committed.
+     * @returns {Promise<string | undefined>} A Promise resolving to the SHA of the commit if successful, otherwise undefined.
+     * @throws Any error that occurs during the execution of the function.
+     */
+                public async updateRekorHost(gitOrg: string, gitRepository: string, rekorHost: string): Promise<string | undefined> {
```
Comment (rhopp) [inline src/apis/git-providers/github.ts:270]: Intentation is off

**Review by rhopp — COMMENTED** [review]
Read through the PR... Found few minor issues (not directly blocking the PR). 
I would like to run the tests tomorrow though ;-)

File: src/apis/git-providers/github.ts (hunk 6 above)
Comment (jkopriva) [reply src/apis/git-providers/github.ts:227]: fixed, thanks!

File: src/apis/git-providers/github.ts (hunk 7 above)
Comment (jkopriva) [reply src/apis/git-providers/github.ts:270]: fixed, thanks!

File: src/apis/git-providers/github.ts [hunk 8]
```diff
@@ -215,6 +216,92 @@ export class GitHubProvider extends Utils {
...
+                console.error("An error occurred while updating the enviroment file:", error);
+            }
+        }
+
+                /**
+     * Enables ACS scan for testing to the main branch of a specified Git repository.
+     * 
+     * @param {string} gitOrg - The name of the GitHub organization.
+     * @param {string} gitRepository - The name of the repository where the file will be committed.
+     * @returns {Promise<string | undefined>} A Promise resolving to the SHA of the commit if successful, otherwise undefined.
```
[the remark this answers is not in the document]
Comment (jkopriva) [reply src/apis/git-providers/github.ts:267]: fixed. thanks!

**Review by rhopp — APPROVED** [review]

**Pull request merged**